owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/shares/ShareType.kt

```diff
@@ -45,33 +45,19 @@
     CONTACT(5),
     FEDERATED(6);
 
-    private int value;
+    companion object {
 
-    private ShareType(int value) {
-        this.value = value;
-    }
-
-    public static ShareType fromValue(int value) {
-        switch (value) {
-            case -1:
-                return NO_SHARED;
-            case 0:
-                return USER;
-            case 1:
-                return GROUP;
-            case 3:
-                return PUBLIC_LINK;
-            case 4:
-                return EMAIL;
-            case 5:
-                return CONTACT;
-            case 6:
-                return FEDERATED;
+        fun fromValue(value: Int): ShareType? {
```

Comment:
Why don't you include an else condition in the when for null values.

This function can be refactored as => 

```
fun fromValue(value: Int): ShareType? =>
        return when (value) {
                -1 -> return NO_SHARED
                0 -> return USER
                1 -> return GROUP
                3 -> return PUBLIC_LINK
                4 -> return EMAIL
                5 -> return CONTACT
                6 -> return FEDERATED
                else -> null
        }
```

Reply:
When I try that code it says `Function 'fromValue' without a body must be abstract` , something missing?

Reply:
I have removed the return statement in each condition. Try and if not, ping me on Skype

Reply:
Solved